crates/acvm/src/pwg/binary.rs

```diff
@@ -82,7 +82,7 @@ impl BinarySolver {
             | GateResolution::UnsatisfiedConstrain
             | GateResolution::UnknownError(_) => return result,
             GateResolution::Skip => (),
-            GateResolution::UnsupportedOpcode(_) | GateResolution::Solved(_) => unreachable!(),
+            GateResolution::UnsupportedOpcode(_) | GateResolution::Solved => unreachable!(),
```

Comment:
Can we remove `Solved` entirely? This case that is marked `unreachable` is the only differing case between it and Resolved anyway

Reply:
The meaning is different so I kept it and instead I fixed prove_cmd.rs so that they are now treated differently.

Reply:
How is the meaning different? I don't think I see enough difference to justify the additional variant. The difference that was added was adding an `unreachable` for the Resolved case and only having the Solved case return a unit. It seems you want to make these mean roughly the same thing but be mutually exclusive in terms of when they can happen. E.g. they could both be replaced with just `Resolved => (),`
One of the reasons I'm pushing back a bit on this is the naming of `Solved` and `Resolved` are quite similar so it is difficult to judge what they actually mean or are meant for.

Reply:
Resolved indicates a gate is resolved, while solved indicates the whole circuit is solved.

Reply:
Alternatively we could split the current `GateResolution` enum into `GateResolution{ Resolved, Skip }` and `GateResolutionError{ UnknownError, UnsatisfiedConstrain, UnsupportedOpcode }`.

`PartialWitnessGenerator.solve` makes more sense returning a `Result<(), GateResolutionError>` in my opinion as `Resolved`/`Skip` aren't meaningful on a circuit level. We could then make use of the `Ok()` value for something else such as returning debug info or the solved witness.

As a bonus, we'd then have a proper error value for solving the circuit (getting rid of the match statement in `solve_witness` which currently would have to copy-pasted by any other users of `PartialWitnessGenerator.solve`)